src/memory/memoryManager.ts

```diff
@@ -1,12 +1,11 @@
-import { Uint32, UnsignedInteger } from 'primitives/uint';
+import { UnsignedInteger } from 'primitives/uint';
 import { Memory } from './memory';
 import { MaybeRelocatable, Relocatable } from 'primitives/relocatable';
-import { Result } from 'result/result';
-import { MemoryError, WriteOnceError } from 'result/memory';
-import { SegmentError } from 'result/primitives';
+import { MemoryError, WriteOnceError } from 'errors/memory';
+import { SegmentError } from 'errors/primitives';
 
 export class MemorySegmentManager {
-  private segmentSizes: Map<Uint32, Uint32>;
+  private segmentSizes: Map<number, number>;
```

Comment:
so I think that this is useless if we use a list of list in the memory

Reply:
Is there an advantage to use list of list? I think the cairo vm in zig was doing list of list, the issue is that you end up reserving real memory for any memory holes, whereas map will just skip the holes.
Example:
[[empty, empty, empty, Relocatable(1,1)]] vs {"0:3": Relocatable(1,1)}